packages/wallets/src/features/cashier/modules/Transfer/components/TransferFormDropdown/TransferFormDropdown.tsx

```diff
@@ -37,9 +37,18 @@ const TransferFormDropdown: React.FC<TProps> = ({ fieldName, label, mobileAccoun
     const selectedAccount = label === 'Transfer from' ? fromAccount : toAccount;
     const accountsList = label === 'Transfer from' ? accounts : toAccountList;
 
-    const handleSelect = (value: TInitialTransferFormValues['fromAccount']) => {
-        setFieldValue(fieldName, value);
-    };
+    const handleSelect = useCallback(
+        (account: TInitialTransferFormValues['fromAccount']) => {
+            if (account?.loginid === selectedAccount?.loginid) return;
+            if (label === 'Transfer from') {
```

Comment:
does not feel good to base application logic on translations/labels, could it use some sort of enum / field ID etc?

Reply:
basing logic on label/translations/strings means that any label/translation change will break the logic

Reply:
mmm
@wojciech-deriv 
does `filedName` can serve as id here?
instead of if (label === 'Transfer from') 
use (filedName === 'fromAccount') ?

Reply:
hey @heorhi-deriv field is okay :) as its unlikely to change, and if its gonna change, then its expected that it will break all the logic related to this specific field (while with translations, if I change them, I do not expect any logic to be broken so its likely to cause bugs)